templates/CMakeLists.txt.template

```diff
@@ -76,7 +76,7 @@
   set(PACKAGE_STRING    "<%text>${PACKAGE_NAME} ${PACKAGE_VERSION}</%text>")
   set(PACKAGE_TARNAME   "<%text>${PACKAGE_NAME}-${PACKAGE_VERSION}</%text>")
   set(PACKAGE_BUGREPORT "https://github.com/grpc/grpc/issues/")
-  project(<%text>${PACKAGE_NAME}</%text> C CXX)
+  project(<%text>${PACKAGE_NAME}</%text> LANGUAGES C CXX VERSION ${settings.core_version})
```

Comment:
core_version is  grpc C core version, but the C++ libraries (grpc++) are at a different version (currently 1.24 ish...)

Reply:
Yes, I'm using both `grpc_VERSION` (8.0.0) and `PACKAGE_VERSION` (1.24.0-dev) to generate the .pc files from CMake. Is there some reason that you would prefer the CMake project version to be 1.24.0-dev instead?

The [VERSION option of the project() command](https://cmake.org/cmake/help/latest/command/project.html#options) sets some CMake variables that you can use elsewhere in your CMakeLists.txt scripts.

Reply:
The public project version is the one used for C++ (see our release notes), the C core version number is irrelevant for many users.

That said, can we keep this PR focused on adding the pkgconfig files and perhaps add the project VERSION option in a separate PR - I'm not generally opposed to setting the project VERSION option, but it seems that it's a bit orthogonal to what this PR is trying to achieve (and there are more things to consider when it comes to setting project VERSION).

Reply:
👍